src/app/ngrx/reducers/index.ts

```diff
@@ -88,5 +96,51 @@ export const selectPodcastMetricsFilteredTotal = createSelector(selectPodcastMet
       return metricsUtil.getTotal(data);
     }
   });
+export const selectPodcastMetricsLoading = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.some((m: PodcastMetricsModel) => m.loading);
+});
+export const selectPodcastMetricsLoaded = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.every((m: PodcastMetricsModel) => m.loaded || m.loaded === undefined);
+});
+const errorType = (code) => code === 401 ? 'Authorization' : 'Unknown';
+export const selectPodcastMetricsError = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.filter(m => m.error).map(m => {
+    return `${errorType(m.error.status)} error occurred while requesting podcast metrics`;
+  });
+});
 
 export const selectEpisodeMetrics = createFeatureSelector<EpisodeMetricsModel[]>('episodeMetrics');
+export const selectEpisodeMetricsLoading = createSelector(selectEpisodeMetrics, (metrics: EpisodeMetricsModel[]) => {
+  return metrics.some((m: EpisodeMetricsModel) => m.loading);
```

Comment:
Ah, I get it.  So the `CastleEpisodeMetricsLoadAction`s are dispatched individually.  But the loading state is across all of those.

Reply:
Yes, so the spinner spins on loading$ until all individual episode metrics are not loading (and episodes and series and podcast metrics.)